extension/scripts/Selector/SelectorTable.js

```diff
@@ -183,41 +256,69 @@ var SelectorTable = {
                 return "thead tr:nth-of-type(" + (rowIndex + 1) + ")";
             }
         }
-        else if (firstRow.find("th:not(:empty)").length > 1) {
-            return "tr:nth-of-type(1)";
-        }
-        else if (firstRow.find("th:first-child:not(:empty)").length == 1 && firstRow.children().length > 1) {
-            return "tr";
-        }
-        else if ($table.find("tr td:not(:empty), tr th:not(:empty)").length) {
-            var $rows = $table.find("tr");
-            // first row with data
-            var rowIndex = $rows.index($rows.filter(":has(td:not(:empty)),:has(th:not(:empty))")[0]);
-            return "tr:nth-of-type(" + (rowIndex + 1) + ")";
-        }
         else {
-            return "";
+            if (!verticalTable) {
+                if (firstRow.find("th:not(:empty)").length > 1) {
+                    return "tr:nth-of-type(1)";
+                } else if (firstRow.find("th:first-child:not(:empty)").length === 1 && firstRow.children().length > 1) {
+                    return "tr";
+                } else if ($table.find("tr td:not(:empty), tr th:not(:empty)").length) {
+                    var $rows = $table.find("tr");
+                    // first row with data
+                    var rowIndex = $rows.index($rows.filter(":has(td:not(:empty)),:has(th:not(:empty))")[0]);
+                    return "tr:nth-of-type(" + (rowIndex + 1) + ")";
+                } else {
+                    return "";
+                }
+            }
+            else {
+                if (firstRow.find("th").length>0) return "tr>th";
+                else return "tr>td:nth-of-type(1)";
+            }
+
+
+
+
+
         }
-    },
 
-    getTableDataRowSelectorFromTableHTML: function (html) {
 
+    },
+    getTableDataRowSelectorFromTableHTML: function (html) {
+        var verticalTable = this.verticalTable;
+        if (this.verticalTable === undefined) {
+            verticalTable = $("#edit-selector [name=verticalTable]").is(":checked");
+        }
         var $table = $(html);
         if ($table.find("thead tr:has(td:not(:empty)), thead tr:has(th:not(:empty))").length) {
```

Comment:
So many nested if expressions is hard to understand. Whe should discuss how to refactor it.